app/controllers/api/v1/projects_controller.rb

```diff
@@ -22,8 +21,38 @@ def create
         end
       end
 
+      def update
+        if update_project.success?
+          render json: { project: @project, message: "Project updated" }
+        else
+          render json: { project: {}, errors: @project.errors }
+        end
+      end
+
+      def destroy
+        if destroy_project.success?
+          render json: { project: @project, message: "Project updated" }
+        else
+          render json: { project: {}, errors: @project.errors }
+        end
+      end
+
       private
 
+      def set_project
+        @project = Project.find_by(id: params[:id])
+      end
+
+      def destroy_project
+        @destroy_project ||=
+          Projects::Destroy.call(project: @project)
+      end
+
+      def update_project
+        @update_project ||=
+          Projects::Update.call(project: @project, project_params: project_params)
+      end
```

Comment:
You already don't really needed memoization here

```suggestion
      def destroy_project
          Projects::Destroy.call(project: @project)
      end

      def update_project
          Projects::Update.call(project: @project, project_params: project_params)
      end
```